common/persistence/persistenceInterface.go

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package persistence
+
+import (
+	"time"
+
+	workflow "github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+)
+
+type (
+
+	//////////////////////////////////////////////////////////////////////
+	// Persistence interface is a lower layer of dataInterface.
+	// The intention is to let different persistence implementation(SQL,Cassandra/etc) share some common logic
+	// Right now the only common part is serialization/deserialization, and only ExecutionManager/HistoryManager need it.
+	// ShardManager/TaskManager/MetadataManager are the same.
+	//////////////////////////////////////////////////////////////////////
+
+	// PersistenceShardManager is a lower level of ShardManager
+	PersistenceShardManager = ShardManager
+	// PersistenceTaskManager is a lower level of TaskManager
+	PersistenceTaskManager = TaskManager
+	// PersistenceMetadataManager is a lower level of MetadataManager
+	PersistenceMetadataManager = MetadataManager
+
+	// PersistenceExecutionManager is used to manage workflow executions for Persistence layer
+	PersistenceExecutionManager interface {
```

Comment:
Let's use Store naming convention here.
ExecutionStore
EventsStore
etc

Reply:
OK sounds good to me.
BTW EventsStore I think you meant HistoryStore